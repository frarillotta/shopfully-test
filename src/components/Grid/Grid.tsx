
import styled from 'styled-components';
import { GridItem } from '..';
import { CompletedFlyer } from '../../types';

type GridProps = {
	flyers: CompletedFlyer[];
	className?: string;
	bookmarkedFlyers: string[],
	addFlyer: (id: number) => void, 
	removeFlyer: (id: number) => void
}

export function Grid({flyers, bookmarkedFlyers, addFlyer, removeFlyer, className = ''}: GridProps) {

	return <Wrapper id={'flyers-grid'} className={className}>
		{flyers.map((flyer) => {
			const {id} = flyer;
			const isBookmarked = bookmarkedFlyers?.includes(String(id)) || false;
			return <GridItem 
				key={id} 
				flyer={flyer} 
				isBookmarked={isBookmarked}
				addFlyer={addFlyer}
				removeFlyer={removeFlyer}
			/>;
		})}
	</Wrapper>;

}


const Wrapper = styled.main`
	display: grid;
	grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
	grid-auto-columns: auto;
	grid-auto-flow: row dense;
	grid-gap: 1rem;
`;