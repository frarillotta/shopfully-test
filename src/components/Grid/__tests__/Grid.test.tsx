import { render } from '@testing-library/react';
import {Grid} from '../Grid';

describe('Grid', () => {

	const mockList = ['mockitem1', 'mockitem2', 'mockitem3'];
	const mockBookmarkedFlyers = ['694041', '456', '693000'];
	const mockFlyers = [{
		'id': 694041,
		'retailer_name': 'mockName',
		'category_name': 'mockName',
		'title': 'Sport acquatici - sup gonfiabile',
		'retailer_id': 3,
		'start_date': '2021-06-01',
		'end_date': '2021-06-30',
		'is_published': 0,
		'asset': 'https://it-it-media.shopfully.cloud/images/volantini/big_694041.jpg',
		'category_id': 7
	},
	{
		'id': 693000,
		'retailer_name': 'mockName',
		'category_name': 'mockName',
		'title': 'Il gusto dellestate.',
		'retailer_id': 23,
		'start_date': '2021-06-10',
		'end_date': '2021-06-20',
		'is_published': 1,
		'asset': 'https://it-it-media.shopfully.cloud/images/volantini/big_693000.jpg',
		'category_id': 6
	},
	{
		'id': 692671,
		'retailer_name': 'mockName',
		'category_name': 'mockName',
		'title': 'Promozione Purina',
		'retailer_id': 15,
		'start_date': '2021-06-03',
		'end_date': '2021-06-30',
		'is_published': 0,
		'asset': 'https://it-it-media.shopfully.cloud/images/volantini/big_692671.jpg',
		'category_id': 5
	}];
	const addFlyer = jest.fn();
	const removeFlyer = jest.fn();
	
	describe('when the grid is rendered', () => {
		
		it('should have a consistent snapshot', () => {
			const {container} = render(<Grid 
				addFlyer={addFlyer} bookmarkedFlyers={mockBookmarkedFlyers} flyers={mockFlyers} removeFlyer={removeFlyer}
			/>
			);
			expect(container).toMatchSnapshot();
		});

	});
	
});