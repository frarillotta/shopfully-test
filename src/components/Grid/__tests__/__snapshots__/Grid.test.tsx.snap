// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid when the grid is rendered should have a consistent snapshot 1`] = `
<div>
  <main
    class="Grid__Wrapper-sc-mkvy52-0 bWukEu"
  >
    <article
      class="GridItem__Wrapper-sc-qnp2w-0 kZQyAV"
      id="flyer-694041"
    >
      <div />
      <div
        class="GridItem__Description-sc-qnp2w-2 cnkKSq"
        id="flyer-description"
      >
        <span
          class="GridItem__RetailerName-sc-qnp2w-3"
          id="flyer-retailer-name"
        >
          mockName
        </span>
        <span
          class="GridItem__Title-sc-qnp2w-4 pJOSk"
          id="flyer-title"
        >
          Sport acquatici - sup gonfiabile
        </span>
        <span
          class="GridItem__CategoryName-sc-qnp2w-5 hfFSve"
          id="flyer-category-name"
        >
          mockName
        </span>
        <button
          class="GridItem__Bookmarked-sc-qnp2w-6 dtrsUa"
          data-t-flyer-is-bookmarked="true"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            style="width: 28px; height: 28px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
        </button>
      </div>
    </article>
    <article
      class="GridItem__Wrapper-sc-qnp2w-0 kZQyAV"
      id="flyer-693000"
    >
      <div />
      <div
        class="GridItem__Description-sc-qnp2w-2 cnkKSq"
        id="flyer-description"
      >
        <span
          class="GridItem__RetailerName-sc-qnp2w-3"
          id="flyer-retailer-name"
        >
          mockName
        </span>
        <span
          class="GridItem__Title-sc-qnp2w-4 pJOSk"
          id="flyer-title"
        >
          Il gusto dellestate.
        </span>
        <span
          class="GridItem__CategoryName-sc-qnp2w-5 hfFSve"
          id="flyer-category-name"
        >
          mockName
        </span>
        <button
          class="GridItem__Bookmarked-sc-qnp2w-6 dtrsUa"
          data-t-flyer-is-bookmarked="true"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            style="width: 28px; height: 28px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
        </button>
      </div>
    </article>
    <article
      class="GridItem__Wrapper-sc-qnp2w-0 kZQyAV"
      id="flyer-692671"
    >
      <div />
      <div
        class="GridItem__Description-sc-qnp2w-2 cnkKSq"
        id="flyer-description"
      >
        <span
          class="GridItem__RetailerName-sc-qnp2w-3"
          id="flyer-retailer-name"
        >
          mockName
        </span>
        <span
          class="GridItem__Title-sc-qnp2w-4 pJOSk"
          id="flyer-title"
        >
          Promozione Purina
        </span>
        <span
          class="GridItem__CategoryName-sc-qnp2w-5 hfFSve"
          id="flyer-category-name"
        >
          mockName
        </span>
        <button
          class="GridItem__Bookmarked-sc-qnp2w-6 foYuNw"
          data-t-flyer-is-bookmarked="false"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            style="width: 28px; height: 28px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
        </button>
      </div>
    </article>
  </main>
</div>
`;
